R/launcher-functions.R

```diff
@@ -0,0 +1,326 @@
+#'   job submission.
+#' @param container The container to be used for launched jobs.
+#' @param exposedPorts The ports that are exposed by services running on a
+#'   container. Only applicable to systems that support containers.
+#' @param mounts A list of mount points. See [launcherHostMount()] and
+#'   [launcherNfsMount()] for more information.
+#' @param placementConstraints A list of placement constraints. See
+#'   [launcherPlacementConstraint()] for more information.
+#' @param resourceLimits A list of resource limits. See [launcherResourceLimit()]
+#'   for more information.
+#' @param queues A list of available submission queues for the cluster. Only
+#'   applicable to batch systems like LSF.
+#'
+#' @param config A list of cluster-specific configuration options. See
+#'   [launcherConfig()] for more information.
+#' @param user The user-name of the job owner.
+#'
+#' @family job submission
+#' @export
+launcherSubmitJob <- function(name,
+                              cluster = "Local",
+                              tags = NULL,
+
+                              command = NULL,
+                              exe = NULL,
+                              args = NULL,
+                              environment = NULL,
+                              stdin = NULL,
+                              stdoutFile = NULL,
+                              stderrFile = NULL,
+                              workingDirectory = NULL,
+
+                              host = NULL,
+                              container = NULL,
+                              exposedPorts = NULL,
+                              mounts = NULL,
+                              placementConstraints = NULL,
+                              resourceLimits = NULL,
+                              queues = NULL,
+
+                              config = NULL,
+                              user = Sys.getenv("USER"))
+{
+  callLauncherFun("launcher.submitJob",
+                  args = args,
+                  cluster = cluster,
+                  command = command,
+                  config = config,
+                  container = container,
+                  environment = environment,
+                  exe = exe,
+                  exposedPorts = exposedPorts,
+                  host = host,
+                  mounts = mounts,
+                  name = name,
+                  placementConstraints = placementConstraints,
+                  queues = queues,
+                  resourceLimits = resourceLimits,
+                  stderrFile = stderrFile,
+                  stdin = stdin,
+                  stdoutFile = stdoutFile,
+                  tags = tags,
+                  user = user,
+                  workingDirectory = workingDirectory)
+}
+
+#' Interact with (Control) a Job
+#'
+#' Interact with a job.
+#'
+#' @param jobId The job id.
+#' @param operation The operation to execute.
+#'
+#' @export
+launcherControlJob <- function(jobId,
```

Comment:
As with resource limits, not all job-control operations are supported by every plug-in.